src/core/utils/event-emitter.ts

```diff
@@ -60,3 +60,12 @@ const InternalEventEmitter: new <EventsMap>() => InterfaceEventEmitter<EventsMap
 class EventEmitter<EventsMap> extends InternalEventEmitter<EventsMap> {}
 
 export default EventEmitter;
+
+/**
+ * Creates a wrapper function that forwards all arguments to the provided function.
+ * @param fn The function to wrap
+ * @returns A new function with the same signature as the input function
+ */
+export function wrap<Args extends unknown[], Return>(fn: (...args: Args) => Return): (...args: Args) => Return {
```

Comment:
Nitpick: can we do this in the other format, e.g.

`export const wrap = (...) => ...`

To be consistent with elsewhere?

Reply:
done